controllers/k8ssandra/k8ssandracluster_controller.go

```diff
@@ -266,6 +270,64 @@ func (r *K8ssandraClusterReconciler) reconcile(ctx context.Context, kc *api.K8ss
 			return ctrl.Result{}, err
 		}
 
+		// reconcile seeds
+		//
+		// The following if block was basically taken straight out of cass-operator. See
+		// https://github.com/k8ssandra/k8ssandra-operator/issues/210 for a detailed
+		// explanation of why this is being done.
+		//
+		// TODO Refactor the if block into a separate method/function.
+		// Note: Some other refactoring needs to be done first so that this can be done cleanly.
+		logger.Info("Reconciling seeds")
+		desiredEndpoints := newEndpoints(desiredDc, seeds)
+		actualEndpoints := &corev1.Endpoints{}
+		endpointsKey := client.ObjectKey{Namespace: desiredEndpoints.Namespace, Name: desiredEndpoints.Name}
+
+		if err = remoteClient.Get(ctx, endpointsKey, actualEndpoints); err == nil {
+			// If we already an Endpoints object but no seeds then it probably means all
+			// Cassandra pods are down or not ready for some reason. In this case we will
+			// delete the Endpoints and let it get recreated once we have seed nodes.
+			if len(seeds) == 0 {
+				logger.Info("Deleting endpoints")
+				if err := remoteClient.Delete(ctx, actualEndpoints); err != nil {
+					logger.Error(err, "Failed to delete endpoints")
+					return ctrl.Result{}, err
+				}
+			} else {
+				if !utils.CompareAnnotations(actualEndpoints, desiredEndpoints, api.ResourceHashAnnotation) {
+					logger.Info("Updating endpoints", "Endpoints", endpointsKey)
+					actualEndpoints := actualEndpoints.DeepCopy()
+					resourceVersion := actualEndpoints.GetResourceVersion()
+					desiredEndpoints.DeepCopyInto(actualEndpoints)
+					actualEndpoints.SetResourceVersion(resourceVersion)
+					if err = remoteClient.Update(ctx, actualEndpoints); err != nil {
+						logger.Error(err, "Failed to update endpoints", "Endpoints", endpointsKey)
+						return ctrl.Result{}, err
+					}
+				}
+			}
+		} else {
+			if errors.IsNotFound(err) {
+				// If we have seeds then we want to go ahead and create the Endpoints. But
+				// if we don't have seeds, then we don't need to do anything for a couple
+				// of reasons. First, no seeds means that cass-operator has not labeled any
+				// pods as seeds which would be the case when the CassandraDatacenter is f
```

Comment:
```suggestion
				// pods as seeds which would be the case when the CassandraDatacenter is
```

Reply:
OK but shouldn't we requeue if endpoints not found and len(seeds) == 0? Is there any value in keeping reconciling the DC without seeds?

Reply:
We have to proceed here; otherwise, we would never create the first CassandraDatacenter.